src/helpers/provider.ts

```diff
@@ -0,0 +1,66 @@
+import providers, { providerLocal } from '../ci_providers'
+import { info, logError, verbose } from '../helpers/logger'
+import { IServiceParams, UploaderInputs } from '../types'
+
+export function detectProvider(inputs: UploaderInputs): IServiceParams {
```

Comment:
🤔

Reply:
It's currently more of a "get service params from detected provider" since it does both. In the context of the other comment, it could be renamed to `getServiceParams` to better reflect how it's used.